configuration/pih/scripts/global/zl.js

```diff
@@ -185,32 +185,35 @@ function setUpObsWithObsDateTime(widgetId){
     }
 }
 
-function returnVisitDateValidation(encounterDate,returnVisitDateLessThanEncounterDateMsg,badReturnVisitDateMsg){
+function setupReturnVisitDateValidation(encounterDate,returnVisitDateLessThanEncounterDateMsg,badReturnVisitDateMsg){
 
-   const domEl= jq("#apptDate").find("input[type=text]");
+   const domEl= getField('apptDate.value');
    const yrRange = encounterDate.getFullYear() + ":" + (new Date().getFullYear() + 1);
-   domEl.datepicker('option','yearRange', yrRange); 
-   domEl.prop("readonly", "readonly"); 
-
-  jq(domEl).change(function(){
-    const nextVisitDate = domEl.datepicker('getDate');
-    const months = (nextVisitDate.getFullYear() - encounterDate.getFullYear()) * 12;
-
-    if(months < 12  && nextVisitDate > encounterDate){
-      jq("#return-visit-date-error-message").text('')
-      jq('.submit').prop("disabled", false);
-    }
-    else if(months == 12 && nextVisitDate.getMonth() <= encounterDate.getMonth()){
-      jq("#return-visit-date-error-message").text('')
-      jq('.submit').prop("disabled", false);
-    }else if(months <= 0 && encounterDate > nextVisitDate){
-        jq("#return-visit-date-error-message").text(returnVisitDateLessThanEncounterDateMsg)
-        jq('.submit').prop("disabled", true);
-    }
-    else{
-      jq("#return-visit-date-error-message").text(badReturnVisitDateMsg)
-      jq('.submit').prop("disabled", true);
-    }
-
-  })        
+   domEl.datepicker('option','yearRange', yrRange);
+   domEl.prop("readonly", "readonly");
+
+   const returnVisitDateValidation = function() {
+     const nextVisitDate = domEl.datepicker('getDate');
+     const months = (nextVisitDate.getFullYear() - encounterDate.getFullYear()) * 12;
+
+     if (months < 12  && nextVisitDate > encounterDate) {
+       getField('apptDate.error').text('').hide();
+       return true;
+     }
+     else if(months == 12 && nextVisitDate.getMonth() <= encounterDate.getMonth()) {
+       getField('apptDate.error').text('').hide();
+       return true;
+     }
+     else if(months <= 0 && encounterDate > nextVisitDate) {
+       getField('apptDate.error').text(returnVisitDateLessThanEncounterDateMsg).show();
+       return false;
+     }
+     else {
+       getField('apptDate.error').text(badReturnVisitDateMsg).show();
+       return false;
+     }
+   }
+
+    jq(domEl).change(returnVisitDateValidation);
+    beforeSubmit.push(returnVisitDateValidation);
```

Comment:
This added this same "returnVisitDateValidation" method to a list of functions that are called when the user tries to submit a form.  This assures that the validation will be checked when the user attempts to save, and works around the issue I described above when/if there are multiple fields that need to validated on a page.

You can see the documentation for how this function works here:
https://wiki.openmrs.org/display/docs/HTML%2BForm%2BEntry%2BJavaScript%2BReference#HTMLFormEntryJavaScriptReference-Definingjavascriptfunctionstocallbeforeformvalidationandsubmission